### PR TITLE
Add `Hide Missing Streams` to video feed settings

### PR DESCRIPTION
**Pull request by LiviaMedeiros**
This adds `Hide Missing Streams` to video feed settings, allowing users to optionally remove cards with `missing` status from their lists.

Most of the time these streams should show up, but in some cases they might bloat feed. For example, setting premiere date to "1 year from now" and deleting waiting room keeps it on channel page 1 the whole time:
hangingwaitingrooms

### CODE REVIEW

File: src/components/video/VideoCardList.vue [hunk 1]
```diff
@@ -128,6 +128,7 @@ export default {
                 hideIgnoredTopics: true,
                 forOrg: "",
                 hidePlaceholder: false,
+                hideMissing: this.$store.state.settings.hideMissing,
```
Comment (LiviaMedeiros) [inline src/components/video/VideoCardList.vue:131]: PTAL on this part, maybe it should be set in different way?
`this.filterConfig` below seems to be empty for channel page, effectively ignoring user settings (e.g. allowing to show placeholder streams even if user disabled them).

Comment (sphinxrave) [reply src/components/video/VideoCardList.vue:131]: hmmm. It's required that hideCollabs is false, and ignoredTopics should also be false, but i guess hidePlaceholder can still abide by the default setting.

**Review by sphinxrave — APPROVED** [review]
let me know if you want me to merge it now or if you want to set hidePlaceholder via store value.

File: src/components/video/VideoCardList.vue (hunk 1 above)
Comment (LiviaMedeiros) [reply src/components/video/VideoCardList.vue]: Makes sense. Would https://github.com/HolodexNet/Holodex/pull/671/commits/c34e5b70652f5dd352602fb2b3cda78c666b0bb0 be a better way then? I.e. respecting all settings by default and overriding only the ones that have to be forced.

File: src/mixins/filterVideos.js [hunk 2]
```diff
@@ -37,7 +37,7 @@ export default {
             }
 
             if (hideMissing) {
-              keep &&= v.status !== "missing";
+                keep &&= v.type !== "missing";
```
Comment (sphinxrave) [inline src/mixins/filterVideos.js:40]: status is correct, type is incorrect.

**Comment by NeloBlivion** [comment]
I feel like default should be `false` to reflect existing behaviour?

**Comment by LiviaMedeiros** [comment]
> I feel like default should be `false` to reflect existing behaviour?

It defaults to `false` here: https://github.com/HolodexNet/Holodex/blob/f1e4c4f64a66f62b93c54e7d3dd1c18e4362691e/src/store/settings.module.js#L25

In https://github.com/HolodexNet/Holodex/blob/f1e4c4f64a66f62b93c54e7d3dd1c18e4362691e/src/components/multiview/VideoSelector.vue#L218-L221 AFAICT it's a MultiView videoselector for live/upcoming videos so this option should be irrelevant (but if for some reason we still got a stream with `.status === 'missing'` there, it should be filtered out).
Or I'm missing something else?

**Comment by NeloBlivion** [comment]
Ah true, fair enough then.

**Comment by sphinxrave** [comment]
Good talk <3

**Pull request merged**